packages/amplify-category-analytics/src/migrations/in-app-messaging-migration.ts

```diff
@@ -1,28 +1,88 @@
 import fs from 'fs-extra';
 import * as path from 'path';
-import { pinpointHasInAppMessagingPolicy, pinpointInAppMessagingPolicyName } from '../utils/pinpoint-helper';
+import { analyticsPush } from '../commands/analytics';
+import { invokeAuthPush } from '../plugin-client-api-auth';
+import { getAllDefaults } from '../provider-utils/awscloudformation/default-values/pinpoint-defaults';
+import { getAnalyticsResources } from '../utils/analytics-helper';
+import {
+  checkIfNotificationsCategoryHasPinpoint,
+  getTemplateMappings,
+  pinpointHasInAppMessagingPolicy,
+  pinpointInAppMessagingPolicyName,
+} from '../utils/pinpoint-helper';
 
 /**
  * checks if the project has been migrated to the latest version of in-app messaging
  */
 export const inAppMessagingMigrationCheck = async (context: $TSContext): Promise<void> => {
-  if (['add', 'update', 'push'].includes(context.input.command) && !pinpointHasInAppMessagingPolicy(context)) {
-    const projectBackendDirPath = pathManager.getBackendDirPath();
+  const projectBackendDirPath = pathManager.getBackendDirPath();
+  const resources = getAnalyticsResources(context);
+
+  if (resources.length > 0 && !pinpointHasInAppMessagingPolicy(context)) {
     const amplifyMeta = stateManager.getMeta();
     const analytics = amplifyMeta[AmplifyCategories.ANALYTICS] || {};
     Object.keys(analytics).forEach(resourceName => {
-      const resourcePath = path.join(projectBackendDirPath, AmplifyCategories.ANALYTICS, resourceName);
-      const templateFilePath = path.join(resourcePath, 'pinpoint-cloudformation-template.json');
+      const analyticsResourcePath = path.join(projectBackendDirPath, AmplifyCategories.ANALYTICS, resourceName);
+      const templateFilePath = path.join(analyticsResourcePath, 'pinpoint-cloudformation-template.json');
       const cfn = JSONUtilities.readJson(templateFilePath);
       const updatedCfn = migratePinpointCFN(cfn);
-      fs.ensureDirSync(resourcePath);
+      fs.ensureDirSync(analyticsResourcePath);
       JSONUtilities.writeJson(templateFilePath, updatedCfn);
     });
   }
+
+  const pinpointApp = checkIfNotificationsCategoryHasPinpoint(context);
+  if (resources.length === 0 && pinpointApp) {
+    const defaultValues = getAllDefaults(context.amplify.getProjectDetails());
+    const notificationsInfo = {
+      appName: pinpointApp.appName,
+      resourceName: pinpointApp.appName,
+    };
+
+    Object.assign(defaultValues, notificationsInfo);
+
+    const resource = defaultValues.resourceName;
+    delete defaultValues.resourceName;
+    const analyticsResourcePath = path.join(projectBackendDirPath, AmplifyCategories.ANALYTICS, resource);
+    fs.ensureDirSync(analyticsResourcePath);
+
+    const parametersFileName = 'parameters.json';
+    const parametersFilePath = path.join(analyticsResourcePath, parametersFileName);
+    const jsonString = JSON.stringify(defaultValues, null, 4);
+    fs.writeFileSync(parametersFilePath, jsonString, 'utf8');
```

Comment:
This can be replaced with a single call to `stateManager.setResourceParametersJson`